File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.py
class Solution:
    def solve(self, idx, num):
        total=0
        while num>0:
            rem=num%10
            num//=10
            total += rem

        if idx == total:
            return True

        return False

    def smallestIndex(self, nums: List[int]) -> int:
        n=len(nums)

        for i in range(n):
            if self.solve(i, nums[i]):
                return i

        return -1

File: 3869-smallest-index-with-digit-sum-equal-to-index/smallest-index-with-digit-sum-equal-to-index.java
class Solution{
    boolean solve(int idx, int num){
        int sum=0;
        while(num>0){
            int rem=num%10;
            num/=10;
            sum += rem;
        }

        if(idx == sum)
            return true;
        return false;
    }

    public int smallestIndex(int[] nums){
        int n=nums.length;

        for(int i=0; i<n; i++){
            if(solve(i, nums[i]))
                return i;
        }

        return -1;
    }
}